internal/clients/http/gcp/gcp_client.go

```diff
@@ -133,8 +150,43 @@ func (c *gcpClient) InsertInstances(ctx context.Context, namePattern *string, im
 
 	op, err := client.BulkInsert(ctx, req)
 	if err != nil {
-		return nil, fmt.Errorf("cannot bulk insert instances: %w", err)
+		span.SetStatus(codes.Error, err.Error())
+		log.Trace().Msgf("Bulk insert operation has failed: %v", err)
+		return nil, nil, fmt.Errorf("cannot bulk insert instances: %w", err)
+	}
+	if err := op.Wait(ctx); err != nil {
+		log.Trace().Msgf("Bulk insert operation has failed: %v", err)
+		span.SetStatus(codes.Error, err.Error())
+		return nil, nil, fmt.Errorf("cannot bulk insert instances: %w", err)
 	}
 
-	return ptr.To(op.Name()), nil
+	filter := fmt.Sprintf("labels.rhhcc-id=%v", uuid)
+	lstReq := &computepb.AggregatedListInstancesRequest{
+		Project: c.auth.Payload,
+		Filter:  &filter,
+	}
+
+	if op.Done() {
+		ids := make([]*string, 0)
+		instances := client.AggregatedList(ctx, lstReq)
+		for {
+			pair, err := instances.Next()
+			if errors.Is(err, iterator.Done) {
+				log.Trace().Msg("Instances iterator has finished")
+				break
+			} else if err != nil {
+				log.Trace().Msgf("An error has occured %s", err.Error())
```

Comment:
Log level. Also I would also try to explain when this happened, e.g. "Error during fetching instance ids" or whatever.

Reply:
also we could use the `.Err(err)` for the error logging.